### ISSUE TITLE
Blazor support JIT-runtime in browser

### ISSUE THREAD

**Issue by ivanivanyuk1993** [dotnet/runtime#45471]
### Is your feature request related to a problem? Please describe.
I am trying to do an app for inside use for enterprise in Blazor - https://fir-ec434.web.app/. While it runs great on desktop, it feels laggy on mobiles. My simple measures show that C# runs ~50 times slower than JS, so I had to move performance critical parts(like url state persisting, which can grow up to O(5000) for every state change which uses UrlReactiveValueProvider or its' ChildValueExtractingReactiveValueProvider-s) to JS

### Describe the solution you'd like
My app doesn't care about initial loading time, it would tolerate loading of whole 200mb JIT-runtime on first load as long as performance is native and long loading happens once and then app uses cache

Hence, my app would benefit from loading performant JIT instead of using interpreter

### Additional context
By the way, described feature would be better than native apps for many use cases, like games
My typical user experience with AppStore and PlayStore games
- Click install
- Wait for load/install, while waiting browse the web or do other stuff
- Close/Swipe from memory AppStore/PlayStore
- Find app in menu
- Click app to open it

For Blazor my expected user experience is:
- Open url
- Wait for load, while waiting browse the web or do other stuff
- Open url with loaded app again

If you are interested in my simple measure, which shows ~50 times performance difference
```cs
            // C# benchmark
            // var a = 0;
            // var s = DateTime.Now;
            // for (var i = 0; i < 1e7; i++)
            // {
            //     ++a;
            // }
            // var f = DateTime.Now;
            // Console.WriteLine((f - s).TotalMilliseconds);
            // Console.WriteLine(a);
            //
            // JS benchmark
            // var a = 0; var s = Date.now(); for (var i = 0; i < 1e7; i++) { ++a }; var f = Date.now(); console.log(f - s); console.log(a);

**Comment by srxqds** [comment]
the browser can support jit mode?

**Comment by ivanivanyuk1993** [comment]
@srxqds 
Well, since clr is written in C++, I think it's not a problem to download it and run it in wasm

**Comment by lewing** [comment]
The WebAssembly runtime is by design much more limited than a normal execution environment and anything approaching just in time compilation for a CLR would be extremely difficult.

**Comment by redradist** [comment]
Is there any work in this area ?

**Comment by danroth27** [comment]
> Is there any work in this area ?

Not currently due to web platform limitations. Our focus instead is on ahead-of-time (AOT) compilation.

**Comment by SupinePandora43** [comment]
> > Is there any work in this area ?
> 
> Not currently due to web platform limitations. Our focus instead is on ahead-of-time (AOT) compilation.

Will we get NativeAOT or NativeAOT-llvm in blazor?

**Comment by danroth27** [comment]
> Will we get NativeAOT or NativeAOT-llvm in blazor?

There are no immediate plans to switch Blazor WebAssembly to NativeAOT. Blazor WebAssembly currently uses the Mono AOT toolchain, the same toolchain used for .NET apps on iOS. Currently the NativeAOT support for WebAssembly is experimental.

**Comment by Thaina** [comment]
> > Will we get NativeAOT or NativeAOT-llvm in blazor?
> 
> There are no immediate plans to switch Blazor WebAssembly to NativeAOT. Blazor WebAssembly currently uses the Mono AOT toolchain, the same toolchain used for .NET apps on iOS. Currently the NativeAOT support for WebAssembly is experimental.

Thanks for the info. Are there issue and roadmap about this experimental? Do we have guideline to enable it?

But actually I am more concern about full JIT support in browser wasm. I can't see why we need AoT in browser when browser actually could support loading and generating code at runtime anyway?

**Comment by danroth27** [comment]
> Are there issue and roadmap about this experimental? Do we have guideline to enable it?

No, experimental means there isn't a committed roadmap for this functionality. See https://github.com/dotnet/runtimelab/blob/feature/NativeAOT-LLVM/docs/using-nativeaot/compiling.md#webassembly or additional details.

> I am more concern about full JIT support in browser wasm. I can't see why we need AoT in browser when browser actually could support loading and generating code at runtime anyway?

From what I understand there is still a limitation in browsers that prevents the runtime generation and loading of WebAssembly modules in a manner suitable for JIT compilation.
